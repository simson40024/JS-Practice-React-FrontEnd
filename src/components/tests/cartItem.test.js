import CartItem from "../cart/cartItem";
import { createRoot } from "react-dom/client";
import { act } from "@testing-library/react";
import * as hooks from "../../hook";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

const state = {
  currency: {
    currency: "UAH",
    rate: 38,
  },
};

const mock = jest.spyOn(hooks, "useAppSelector");
const mockDispatch = jest.spyOn(hooks, "useAppDispatch");

describe("CartItem", () => {
  it("should create CartItem", () => {
    mock.mockImplementation((selector) => selector(state));
    const total = 150;
    act(() => {
      createRoot(container).render(
        <CartItem
          id="12345"
          name="cool T-shot"
          price={10}
          quantity={2}
          imgUrl="./img.jpg"
        />
      );
    });
    expect(container).toMatchSnapshot();
  });
});
