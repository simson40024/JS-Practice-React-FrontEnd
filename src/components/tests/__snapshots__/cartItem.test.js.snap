// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should create CartItem 1`] = `
<div>
  <div
    class="cart-row"
    id="12345"
  >
    <div
      class="cart-product"
    >
      <img
        alt=""
        src="./img.jpg"
        width="150px"
      />
      <p>
        cool T-shot
      </p>
    </div>
    <div
      class="cart-price"
    >
      <p>
        380
         
        <br
          class="hidden-br"
        />
        UAH
      </p>
    </div>
    <div
      class="product-quantity"
    >
      <button
        role="button"
      >
        -
      </button>
      <input
        readonly=""
        type="text"
        value="2"
      />
      <button
        role="button"
      >
        +
      </button>
    </div>
    <div
      class="cart-total"
    >
      <p>
        760
         
        <br
          class="hidden-br"
        />
        UAH
      </p>
    </div>
    <div
      class="cart-button"
    >
      <button
        title="delete produrt"
      >
        X
      </button>
    </div>
  </div>
</div>
`;
